### PR TITLE
Created Working protype 

### PR DESCRIPTION
**Pull request by codepoet2017390**
Created Working prototype that successfully registers patient via API using only name.
Please build and test this version of App conforming that the patient is registered on the Staging server.

### CODE REVIEW

**Comment by HerbertYiga** [comment]
could you be having some snap shots of the ui for this?

**Comment by codepoet2017390** [comment]
Hi @HerbertYiga so good to see community involvement in the project.
Regarding snapshots of UI did you take a look at our project Readme. It has some screenshots of the app.

File: app/src/main/java/com/example/nigeriatelemedicineapp/api/ApiManager.kt [hunk 1]
```diff
@@ -22,9 +23,12 @@ class ApiManager {
 
     fun createService()
     {
+        val OkHttpClient= OkHttpInterceptor().getClient("admin","Admin123")
```
Comment (isears) [inline app/src/main/java/com/example/nigeriatelemedicineapp/api/ApiManager.kt]: Let's move these values into static final class variables. Eventually, they should probably be configuration values, but I know we're still figuring out how we want to structure our configs.

File: app/src/main/java/com/example/nigeriatelemedicineapp/api/OkHttpInterceptor.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                    .header("Authorization", basic)
+                    .header("Accept", "application/json")
+                    .method(original.method(), original.body())
+
+
+                val request = requestBuilder.build()
+                chain.proceed(request)
+            }
+            //val logging = HttpLoggingInterceptor()
+            //logging.level = HttpLoggingInterceptor.Level.BODY
```
Comment (isears) [inline app/src/main/java/com/example/nigeriatelemedicineapp/api/OkHttpInterceptor.kt]: Make sure to keep commented-out code out of the repository unless there's a good justification for it.

File: app/src/main/java/com/example/nigeriatelemedicineapp/api/services/RegisterPatientService.kt [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
-
-
-
-
-
-
+    @Headers(
+        "accept: application/json",
+        "content-type: application/json"
+        //"authorization: Basic YWRtaW46QWRtaW4xMjM="
```
Comment (isears) [inline app/src/main/java/com/example/nigeriatelemedicineapp/api/services/RegisterPatientService.kt]: Let's get rid of this too, unless there's a good justification for it

**Review by isears — CHANGES_REQUESTED** [review]
When I run this in my local emulator, I get "Error fetching identifier". Is this expected behavior at this point?

**Comment by codepoet2017390** [comment]
@isears please make sure your emulator has access to the internet.

File: app/src/main/java/com/example/nigeriatelemedicineapp/api/ApiManager.kt (hunk 1 above)
Comment (codepoet2017390) [reply app/src/main/java/com/example/nigeriatelemedicineapp/api/ApiManager.kt]: @isears  Solved

File: app/src/main/java/com/example/nigeriatelemedicineapp/api/OkHttpInterceptor.kt (hunk 2 above)
Comment (codepoet2017390) [reply app/src/main/java/com/example/nigeriatelemedicineapp/api/OkHttpInterceptor.kt]: @isears solved

File: app/src/main/java/com/example/nigeriatelemedicineapp/api/services/RegisterPatientService.kt (hunk 3 above)
Comment (codepoet2017390) [reply app/src/main/java/com/example/nigeriatelemedicineapp/api/services/RegisterPatientService.kt]: @isears yup that was redundant

**Review by isears — APPROVED** [review]

**Pull request merged**